pandas/tests/test_expressions.py

```diff
@@ -442,3 +442,19 @@ def test_bool_ops_warn_on_arithmetic(self):
                     r = f(df, True)
                     e = fe(df, True)
                     tm.assert_frame_equal(r, e)
+
+    def test_bool_ops_column_name_dtype(self):
+        # GH 22383 - .ne fails if columns containing column name 'dtype'
+        df_has_error = DataFrame([[0, 1, 2, 'aa'], [0, 1, 2, 'aa'],
+                                  [0, 1, 5, 'bb'], [0, 1, 5, 'bb'],
+                                  [0, 1, 5, 'bb'], ['cc', 4, 4, 4]],
+                                 columns=['a', 'b', 'c', 'dtype'])
+        df = DataFrame([[0, 1, 2, 'aa'], [0, 1, 2, 'aa'],
+                        [0, 1, 5, 'bb'], [0, 1, 5, 'bb'],
+                        [0, 1, 5, 'bb'], ['cc', 4, 4, 4]],
+                       columns=['a', 'b', 'c', 'd'])
+        result = df_has_error.loc[:, ['a', 'dtype']].ne(df_has_error.loc[:,
+                                                        ['a', 'dtype']])
```

Comment:
make an expected= line

instead of calling the function on a good df, just construct the resulting dataframe directly